Watchman.DomainModel/Warns/Queries/Handlers/GetWarnEventsQueryHandler.cs

```diff
@@ -16,11 +18,16 @@ public GetWarnEventsQueryHandler(ISessionFactory sessionFactory)
         public GetWarnEventsQueryResults Handle(GetWarnEventsQuery query)
         {
             using var session = this._sessionFactory.Create();
-            var filteredEvents = session.Get<WarnEvent>().Where(x => x.ReceiverId == query.UserId);
+            var filteredEvents = session.Get<WarnEvent>();
+            if (query.ReceiverId != 0)
+            {
+                filteredEvents = filteredEvents.Where(x => x.ReceiverId == query.ReceiverId);
+            }
             if (query.ServerId != 0)
             {
                 filteredEvents = filteredEvents.Where(x => x.ServerId == query.ServerId);
             }
+            filteredEvents = filteredEvents.Where(x => x.CreatedAt >= query.From && x.CreatedAt <= query.To);
```

Comment:
myśle że sprawdzanie po CreatedAt powinno być na samym początku - bo co jeśli mamy dane z kilku lat
a chcemy sprawdzać jedynie ostatni miesiąc?
chociaż w sumie
jeśli jest mało warnów per osoba
to jak odfiltrujesz po osobie
to będzie bardzo mało wyników
@gagyn co myślisz?

poza tym
myśle że można użyć do tego obiektu TimeRange i dodać mu metode sprawdzającą czy konkretna data mieści sie w przedziale czasowym 
np 
```csharp
public bool Contains(Datetime dateTime)
{
   return this.Start <= dateTime && this.End >= dateTime;
}
```

Reply:
Nie mogę w linq użyć tej metody niestety

Reply:
w jakim sensie?
filteredEvents.Where(x => query.TimeRange.Contains(x.CreatedAt))
to nie działa?

Reply:
tak, leci exception jak iterować foreachem wynik tej operacji

Reply:
to coś tutaj jest nie tak
jaki exception?

Reply:
![111](https://user-images.githubusercontent.com/47067171/97790706-1c2db280-1bcb-11eb-9da4-30b9663b9c99.jpg)

Kiedy na miejscu porównuję a nie przez TimeRange to działa

Reply:
dobra zostaw już tak jak jest